chromium_src/net/url_request/url_request_http_job.cc

```diff
@@ -41,7 +68,27 @@ bool ShouldUseEphemeralStorage(net::URLRequestHttpJob* http_job) {
   else
 
 #define BRAVE_SAVECOOKIESANDNOTIFYHEADERSCOMPLETE                              \
-  if (ShouldUseEphemeralStorage(this))                                         \
+  bool use_ephemeral_storage = ShouldUseEphemeralStorage(request_);            \
+  if (use_ephemeral_storage) {                                                 \
+    auto top_frame_url =                                                       \
+        request_->isolation_info().top_frame_origin()->GetURL();               \
+    auto cookie_1p = net::CanonicalCookie::CreateSanitizedCookie(              \
+        top_frame_url, cookie->Name(), cookie->Value(), cookie->Domain(),      \
+        cookie->Path(), cookie->CreationDate(), cookie->ExpiryDate(),          \
+        cookie->LastAccessDate(), cookie->IsSecure(), cookie->IsHttpOnly(),    \
+        cookie->SameSite(), cookie->Priority(), cookie->IsSameParty());        \
+                                                                               \
+    net::CookieOptions::SameSiteCookieContext same_site_context =              \
+        net::cookie_util::ComputeSameSiteContextForResponse(                   \
+            top_frame_url, request_->site_for_cookies(),                       \
+            request_->initiator(), false);                                     \
+    net::CookieOptions options_1p = CreateCookieOptions(same_site_context);    \
+                                                                               \
+    if (!(CanSetCookie(*cookie_1p, &options_1p) &&                             \
+          !CanSetCookie(*cookie, &options)))                                   \
+      use_ephemeral_storage = false;                                           \
+  }                                                                            \
```

Comment:
I think you can avoid duplicating this code by making it part of `ShouldUseEphemeralStorage` and passing in the `cookie` parameter.

Reply:
yep, somewhere in here I left a TODO to clean this up

Reply:
yea, I think there are several places this can be cleaned up so I'll create some follow-up issues